community/server/src/main/java/org/neo4j/server/rrd/RrdFactory.java

```diff
@@ -71,28 +69,37 @@ public RrdFactory( Config config, Logging logging )
         this.log = logging.getConsoleLog( getClass() );
     }
 
-    public org.neo4j.server.database.RrdDbWrapper createRrdDbAndSampler( final Database db, JobScheduler scheduler ) throws IOException
+    public org.neo4j.server.database.RrdDbWrapper createRrdDbAndSampler( final Database db, JobScheduler scheduler )
+            throws IOException
     {
         NeoStoreProvider neoStore = db.getGraph().getDependencyResolver().resolveDependency( NeoStoreProvider.class );
+        SampleableDatabaseGuard sampleableDatabaseGuard = new SampleableDatabaseGuard();
+        db.getGraph().getDependencyResolver().resolveDependency( AvailabilityGuard.class ).addListener( sampleableDatabaseGuard );
 
         Sampleable[] primitives = {
-                new NodeIdsInUseSampleable( neoStore ),
-                new PropertyCountSampleable( neoStore ),
-                new RelationshipCountSampleable( neoStore )
+                new NodeIdsInUseSampleable( neoStore, sampleableDatabaseGuard ),
+                new PropertyCountSampleable( neoStore, sampleableDatabaseGuard ),
+                new RelationshipCountSampleable( neoStore, sampleableDatabaseGuard )
         };
 
         Sampleable[] usage = {};
 
+        File oldRrdFile = new File( db.getGraph().getStoreDir(), "rrd" );
```

Comment:
well it's not that important i agree, but we will probably have couple of 2.2 versions and migrations should be separated from main code path.